homeassistant/components/modem_callerid/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for Modem Caller ID integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from phone_modem import PhoneModem
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_DEVICE, DEFAULT_NAME, DOMAIN, EXCEPTIONS
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"name": str, "device": str})
+
+
+class PhoneModemFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Phone Modem."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            device = user_input[CONF_DEVICE]
+
+            await self.async_set_unique_id(device)
```

Comment:
I'm personally not too sure if we should make the serial port the unique ID. What if the port changes? I can see a possibility for the user to reconfigure it later? (or in the future).

Reply:
I would use `_abort_if_unique_id_configured(updates=[CONF_DEVICE: "serial"])` so we don't have duplicate entries which is easy to do by accident. Hot plugging the modem assigns a new device path. I'm assuming that I can get a serial number with `usb`, right?

Reply:
I'm not sure if we can reliably can access the serial number in Python. My comment merely touched upon the fact that the path probably isn't a stable unique ID